File: tests/libjournal/file.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <j2/journal.h>
// support
#include <cassert>
#include <fstream>
#include <filesystem>


// alias the type
using stream_t = pyre::journal::stream_t;


// exercise the trivial device
int main() {
    // the path of the file
    auto filename = std::filesystem::path("file.out");

    // make a file stream
    auto ofs = std::ofstream(filename);
    // instantiate
    stream_t stream(filename, ofs);
    // check its name
    assert (stream.name() == filename);

    // clean up
    // close the file
    ofs.close();
    // and remove it
    std::filesystem::remove(filename);

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/file.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Verify that we can build a stream based device
    """
    # pick a file name
    filename = "file.out"
    # make a stream
    ostr = open(filename, mode="w")

    # get the device
    from j2.Stream import Stream as stream
    # instantiate
    device = stream(name=filename, stream=ostr)
    # check its name
    assert device.name == filename

    # close the stream; not strictly necessary, but let's exercise the interface
    device.close()

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
